Check for dlopen in the libc too.
Some systems (eg FreeBSD) provide it there.

--- mozilla-1.9.2/configure.in.old	2010-01-11 00:15:26 +0100
+++ mozilla-1.9.2/configure.in	2010-01-11 00:27:10 +0100
@@ -3129,10 +3129,15 @@
     ;;
 *)
     AC_CHECK_LIB(m, atan)
-    AC_CHECK_LIB(dl, dlopen,
-    AC_CHECK_HEADER(dlfcn.h, 
-        LIBS="-ldl $LIBS"
-        AC_DEFINE(HAVE_LIBDL)))
+    ac_cv_my_have_libdl=no
+    AC_CHECK_FUNC(dlopen,
+        ac_cv_my_have_libdl=yes,
+        AC_CHECK_LIB(dl, dlopen,
+            ac_cv_my_have_libdl=yes
+            LIBS="-ldl $LIBS"))
+    if test "${ac_cv_my_have_libdl}" = "yes"; then
+    	AC_CHECK_HEADER(dlfcn.h, AC_DEFINE(HAVE_LIBDL))
+    fi
     ;;
 esac
 
